src/hotspot/share/opto/compile.cpp

```diff
@@ -4366,7 +4372,7 @@ Compile::TracePhase::TracePhase(const char* name, elapsedTimer* accumulator)
 }
 
 Compile::TracePhase::~TracePhase() {
-  if (_compile->failing()) return;
+  if (_compile->failing_internal()) return; // timing code, not stressing bailouts.
```

Comment:
While at it, I suggest to add braces:

```suggestion
  if (_compile->failing_internal()) {
    return; // timing code, not stressing bailouts.
  }
```
Same below at some places.